larpandoracontent/LArReclustering/CheatedThreeDClusteringTool.cc

```diff
@@ -23,72 +23,36 @@ CheatedThreeDClusteringTool::CheatedThreeDClusteringTool()
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
-bool CheatedThreeDClusteringTool::Run(const Algorithm *const pAlgorithm, std::vector<pandora::CaloHitList*> &newCaloHitListsVector)
+std::vector<std::reference_wrapper<pandora::CaloHitList>> CheatedThreeDClusteringTool::Run(const Algorithm *const /*pAlgorithm*/, std::reference_wrapper<pandora::CaloHitList> &inputCaloHitList)
```

Comment:
I think I would still be inclined to argue that the reference wrapper is not necessary here (is there a particular example involving the pointers that you based this on?). I think this function is not inherited from any base class you don't control? This comment would apply everywhere else the reference wrapper solution was applied.

I think you are fine with `const CaloHitList &inputCaloHitList` etc (note you can also drop the pandora namespace prefix outside of header files). With the changes you've already made I think this would result in minimal code changes (I think the occasional `get()` function call is the only usage difference?). In addition, you could make the returned reference wrapper an output parameter via an equivalent `std::vector<CaloHitList> &inputCaloHitList` (which again, I think would have fairly limited impact on code structure). I think the original return value of bool is fine, the key difference here is that your input list would now be a constant, and you would have a new output list that is passed by reference from the calling function(I think that's more consistent with other tool usage such as the various cosmic ray tools in ThreeDReco), I'll tag @absolution1 re this as well.